src/app/content/highlights/components/SummaryPopup/FiltersList.tsx

```diff
@@ -0,0 +1,161 @@
+  overflow: hidden;
+  text-overflow: ellipsis;
+  text-transform: capitalize;
+`;
+
+interface FiltersListColorItemProps {
+  type: 'colors';
+  color: HighlightColorEnum;
+  onRemoveItem: (type: 'colors', color: HighlightColorEnum) => void;
+}
+
+interface FiltersListChapterItemProps {
+  type: 'chapters';
+  title: string;
+  chapterId: string;
+  onRemoveItem: (type: 'chapters', chapterId: string) => void;
+}
+
+type FiltersListItemProps = FiltersListColorItemProps | FiltersListChapterItemProps;
+
+// tslint:disable-next-line: variable-name
+const FiltersListItem = (props: FiltersListItemProps) => {
+  const handleClick = () => {
+    if (props.type === 'colors') {
+      props.onRemoveItem(props.type, props.color);
+    } else {
+      props.onRemoveItem(props.type, props.chapterId);
+    }
+  };
+
+  let body: JSX.Element | string;
+  if (props.type === 'colors') {
+    body = <FormattedMessage id={`i18n:highlighting:colors:${props.color}`}>
+      {(msg: Element | string) => msg}
+    </FormattedMessage>;
+  } else {
+    body = <span dangerouslySetInnerHTML={{ __html: props.title }}/>;
+  }
+
+  return <li>
+    <RemoveIcon onClick={handleClick}><Times /></RemoveIcon>
+    <ItemLabel>{body}</ItemLabel>
+  </li>;
+};
+
+interface FiltersListProps {
+  className?: string;
+}
+
+type SectionsMap = Map<string, LinkedArchiveTree | LinkedArchiveTreeSection>;
+
+// tslint:disable-next-line: variable-name
+const FiltersList = ({className}: FiltersListProps) => {
+  const [sections, setSections] = React.useState<SectionsMap>(new Map());
+
+  const book = useSelector(bookSelector);
+  const selectedChapters = useSelector(chaptersFilter);
+  const selectedColors = useSelector(colorsFilter);
+
+  const dispatch = useDispatch();
+
+  const setSelectedChapters = (chapterIds: string[]) => {
+    dispatch(setChaptersFilter(chapterIds));
+    dispatch(filtersChange());
+  };
+  const setSelectedColors = (colors: HighlightColorEnum[]) => {
+    dispatch(setColorsFilter(colors));
+    dispatch(filtersChange());
+  };
+
+  const onRemoveItem = (type: 'colors' | 'chapters', label: HighlightColorEnum | string) => {
+    if (type === 'colors') {
+      setSelectedColors(selectedColors.filter(not(match(label))));
+    } else if (type === 'chapters') {
+      setSelectedChapters(selectedChapters.filter(not(match(label))));
+    }
+  };
+
+  React.useEffect(() => {
+    if (book) {
+      const newSections = new Map(
+        flattenArchiveTree(book.tree).filter((section) =>
+          (section.parent && archiveTreeSectionIsBook(section.parent))
+          || archiveTreeSectionIsChapter(section)).map((s) => [s.id, s])
+        );
+      setSections(newSections);
+    }
+  }, [book]);
+
+  return <ul className={className}>
+    {selectedChapters.map((chapterId) => sections.has(chapterId) && <FiltersListItem
+      key={chapterId}
+      type='chapters'
+      title={sections.get(chapterId)!.title}
+      chapterId={chapterId}
+      onRemoveItem={onRemoveItem}
```

Comment:
you're already assigning this separately for chapters and colors, i'm not sure what value the shared onRemoveItem function is providing, you're merging the data flow just to have to split it out again.